mission/joystick_interface_auv/joystick_interface_auv/joystick_interface_auv.py

```diff
@@ -0,0 +1,239 @@
+        self.get_logger().info(
+            "Joystick interface is up and running. \n When the XBOX controller is connected, press the killswitch button once to enter XBOX mode."
+        )
+
+        self.last_button_press_time_ = 0
+        self.debounce_duration_ = 0.25
+        self.state_ = States.NO_GO
+
+        self.joystick_buttons_map_ = [
+            "A",
+            "B",
+            "X",
+            "Y",
+            "LB",
+            "RB",
+            "back",
+            "start",
+            "power",
+            "stick_button_left",
+            "stick_button_right",
+        ]
+
+        self.joystick_axes_map_ = [
+            "horizontal_axis_left_stick",  #Translation (Left and Right)
+            "vertical_axis_left_stick",  #Translation (Forwards and Backwards)
+            "LT",  #Negative thrust/torque multiplier
+            "horizontal_axis_right_stick",  #Rotation
+            "vertical_axis_right_stick",
+            "RT",  #Positive thrust/torque multiplier
+            "dpad_horizontal",
+            "dpad_vertical",
+        ]
+
+        self.joy_subscriber_ = self.create_subscription(
+            Joy, "joystick/joy", self.joystick_cb, 1)
+        self.wrench_publisher_ = self.create_publisher(Wrench,
+                                                       "thrust/wrench_input",
+                                                       1)
+
+        self.declare_parameter('surge_scale_factor', 0.0)
+        self.declare_parameter('sway_scale_factor', 0.0)
+        self.declare_parameter('yaw_scale_factor', 0.0)
+        self.declare_parameter('heave_scale_factor', 0.0)
+        self.declare_parameter('roll_scale_factor', 0.0)
+        self.declare_parameter('pitch_scale_factor', 0.0)
+
+        #Gets the scaling factors from the yaml file
+        self.joystick_surge_scaling_ = self.get_parameter(
+            'surge_scale_factor').value
+        self.joystick_sway_scaling_ = self.get_parameter(
+            'sway_scale_factor').value
+        self.joystick_yaw_scaling_ = self.get_parameter(
+            'yaw_scale_factor').value
+        self.joystick_heave_scaling_ = self.get_parameter(
+            'heave_scale_factor').value
+        self.joystick_roll_scaling_ = self.get_parameter(
+            'roll_scale_factor').value
+        self.joystick_pitch_scaling_ = self.get_parameter(
+            'pitch_scale_factor').value
+
+        #Killswitch publisher
+        self.software_killswitch_signal_publisher_ = self.create_publisher(
+            Bool, "softWareKillSwitch", 10)
+        self.software_killswitch_signal_publisher_.publish(
+            Bool(data=False))  #Killswitch is not active
+
+        #Operational mode publisher
+        self.operational_mode_signal_publisher_ = self.create_publisher(
+            Bool, "softWareOperationMode", 10)
+
+        # Signal that we are not in autonomous mode
+        self.operational_mode_signal_publisher_.publish(Bool(data=True))
```

Comment:
softWareOperationMode being True does not really say much. Consider using something other than Bool.